### ISSUE TITLE
Switch to tempfile instead of tempdir

### ISSUE THREAD

**Issue by emoon** [emoon/dynamic_reload#25]
This is because tempdir has been deprecated with this info from cargo-deny

```
31 │ tempdir 0.3.7 registry+https://github.com/rust-lang/crates.io-index
   │ ------------------------------------------------------------------- unmaintained advisory detected
   │
   = ID: RUSTSEC-2018-0017
   = Advisory: https://rustsec.org/advisories/RUSTSEC-2018-0017
   = The [`tempdir`](https://crates.io/crates/tempdir) crate has been deprecated
     and the functionality is merged into [`tempfile`](https://crates.io/crates/tempfile).
   = Announcement: https://github.com/rust-lang-deprecated/tempdir/pull/46
   = Solution: No safe upgrade is available!
   = tempdir v0.3.7
     └── dynamic_reload v0.8.0
         └── rv_core v0.1.0
```

**Comment by roy-work** [comment]
[RUSTSEC-2023-0018](https://rustsec.org/advisories/RUSTSEC-2023-0018.html), as well. (`remove_dir_all` is a dependency of `tempdir`.)

**Comment by emoon** [comment]
Fixed in 0.9.0
